<?xml version="1.0"?>
<!--
/****************************************************************************
 *   Copyright (c) 2017 Michael Shomin. All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 *
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in
 *    the documentation and/or other materials provided with the
 *    distribution.
 * 3. Neither the name ATLFlight nor the names of its contributors may be
 *    used to endorse or promote products derived from this software
 *    without specific prior written permission.
 *
 * NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS LICENSE.
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
 * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
 * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
 * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
 * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
 * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
 * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
 * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGE.
 * In addition Supplemental Terms apply.  See the SUPPLEMENTAL file.
 ****************************************************************************/
-->

<launch>
  <arg name="cam_name" default="downward"/>
  <arg name="nodelet_manager_name" default="downward_nodelet_manager"/>
  <arg name="skip_n_frames" default="0"/>
  <arg name="do_rectification" default="false"/>

  <!-- Note that this feature is only applicable to 8096 (if false)-->
  <arg name="is_cam_master" default="true"/>

  <group ns="$(arg cam_name)">
    <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" output="screen"/>

    <!-- camera driver nodelet -->
    <node pkg="nodelet" type="nodelet" name="snap_cam_nodelet"
    args="load snap_cam_ros/driver $(arg nodelet_manager_name)">
      <param name="camera_name" value="$(arg cam_name)"/>
      <!-- Use format=0 for 8096 or format=1 for 8074 -->
      <param name="format" value="0"/>
      <param name="camera_number" value="1"/>
      <param name="width" value="640"/>
      <param name="height" value="480"/>
      <param name="frame_id" value="dfc"/>
      <param name="skip_n_frames" value="$(arg skip_n_frames)"/>
      <param name="is_cam_master" value="$(arg is_cam_master)"/>
    </node>
    <param name="image_raw/compressed/jpeg_quality" value="50"/>
    <!-- mono rectification -->
    <group if="$(arg do_rectification)">
      <node pkg="nodelet" type="nodelet" name="image_proc_rect"
      args="load image_proc/rectify $(arg nodelet_manager_name)">
        <remap from="image_mono" to="image_raw"/>
      </node>
    </group>

  </group>
</launch>
